src/Domain/DonationRepository.php

```diff
@@ -210,9 +215,12 @@ public function releaseMatchFunds(Donation $donation): void
         // were able to observe. This is why we now go straight to trying to `acquire()`. If
         // this returns false, we are in the contested lock case and know to drop this attempt.
 
-        $fundsReleaseLock = $this->lockFactory->createLock("release-funds-{$donation->getUuid()}");
+        $fundsReleaseLock = $this->getFundsReleaseLock($donation);
 
         try {
+            // We don't `release()` the lock, holding it for the default 5 mins instead unless a thread newly
```

Comment:
where does 5 minutes come from? Maybe worth adding a reference to that in the comment.